### PR TITLE
Add "copy cURL" to Network Monitor

### PR DESCRIPTION
**Pull request by jonbell-lot23**
Addresses https://linear.app/replay/issue/FE-826/network-monitor-copy-as-curl

image

image

### CODE REVIEW

**Comment by jonbell-lot23** [comment]
There are typescript errors I need to work through, stay tuned

File: src/ui/components/NetworkMonitor/RequestTable.tsx [hunk 1]
```diff
@@ -78,73 +81,100 @@ const RequestTable = ({
     }
   };
 
+  const copyAsCurl = () => {
+    setContextMenuData(null);
+    navigator.clipboard.writeText("curl '" + contextMenuData.row.values.url + "'");
```
Comment (jasonLaster) [inline src/ui/components/NetworkMonitor/RequestTable.tsx:86]: I think the curl also includes header information as well.

**Review by jasonLaster — COMMENTED** [review]
Very cool

File: src/ui/components/NetworkMonitor/RequestTable.tsx [hunk 2]
```diff
@@ -78,73 +81,100 @@ const RequestTable = ({
...
+
+    setShowCopied(true);
+    setTimeout(() => setShowCopied(false), 2700);
+    setTimeout(() => document.getElementById("showCopied").classList.remove("opacity-100"), 2000);
+    setTimeout(() => document.getElementById("showCopied").classList.add("opacity-0"), 2000);
```
Comment (bvaughn) [inline src/ui/components/NetworkMonitor/RequestTable.tsx:91]: The way this is written (a chain of `setTimeout` calls that attempt to line up with a CSS transition) seems kind of brittle. If it's the effect you're going for, I don't have a better suggestion, but maybe we can make it a little more obvious what it's doing?

```suggestion
let timeout = setTimeout(() => {
  document.getElementById("showCopied").classList.remove("opacity-100");
  document.getElementById("showCopied").classList.add("opacity-0");
  
  // The fade out transition lasts 700ms
  timeout = setTimeout(() => {
    setShowCopied(false);
  }, 700);
}, 2000);

return () => {
  clearTimeout(timeout);
}
```

Also we should probably clear the timeout on unmount (as shown above) so we don't risk a potential undefined access error.

Comment (jonbell-lot23) [reply src/ui/components/NetworkMonitor/RequestTable.tsx:91]: Thanks! All good suggestions.

**Comment by jonbell-lot23** [comment]
This item was moved into the FE backlog for now.

**Pull request closed without merging**